Source/LinqToDB/Data/DataConnection.Async.cs

```diff
@@ -26,18 +26,28 @@ public virtual async Task<DataConnectionTransaction> BeginTransactionAsync(Cance
 			//
 			if (TransactionAsync != null) await TransactionAsync.DisposeAsync().ConfigureAwait(Common.Configuration.ContinueOnCapturedContext);
 
-			// Create new transaction object.
-			//
-			TransactionAsync = await _connection!.BeginTransactionAsync(cancellationToken).ConfigureAwait(Common.Configuration.ContinueOnCapturedContext);
+			var dataConnectionTransaction = await TraceActionAsync(
+				this,
+				TraceOperation.BeginTransaction,
+				"BeginTransaction",
+				static async (dataConnection, _, cancellationToken) =>
+				{
+					// Create new transaction object.
+					//
+					dataConnection.TransactionAsync = await dataConnection._connection!.BeginTransactionAsync(cancellationToken).ConfigureAwait(Common.Configuration.ContinueOnCapturedContext);
 
-			_closeTransaction = true;
+					dataConnection._closeTransaction = true;
 
-			// If the active command exists.
-			//
-			if (_command != null)
-				_command.Transaction = Transaction;
+					// If the active command exists.
+					//
+					if (dataConnection._command != null)
+						dataConnection._command.Transaction = dataConnection.Transaction;
 
-			return new DataConnectionTransaction(this);
+					return new DataConnectionTransaction(dataConnection);
+				}, (object?)null, cancellationToken)
```

Comment:
Slightly preferred according to newer style standards, but no real preference either way.

```suggestion
				}, default(object?), cancellationToken)
```